cura/API/Account.py

```diff
@@ -1,15 +1,18 @@
 # Copyright (c) 2018 Ultimaker B.V.
 # Cura is released under the terms of the LGPLv3 or higher.
-from typing import Optional, Dict
+from typing import Optional, Dict, TYPE_CHECKING
 
 from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot, pyqtProperty
 
+from UM.i18n import i18nCatalog
 from UM.Message import Message
+
 from cura.OAuth2.AuthorizationService import AuthorizationService
 from cura.OAuth2.Models import OAuth2Settings
-from UM.Application import Application
 
-from UM.i18n import i18nCatalog
+if TYPE_CHECKING:
+    from cura.CuraApplication import CuraApplication
```

Comment:
This is really a symptom that you are passing around far more than necessary. The `Account` class takes the whole application object when all it needed was some preference and settings data. That is a massive amount of unneeded data to share around. It also makes reasoning about what the `Account` class can and cannot do from the outside really hard. (It could do anything it likes, it has the whole application object to play with!). It is not clear from a higher level what other subsystems the `Account` class talks to.

Passing around a whole application object is not a good thing.

Reply:
I agree, there's still too many dependencies here. But giving specific objects such as `Preferences` doesn't solve the problem either, because those dependencies are still there. IMO, this is a problem that should be solved in a Cura and/or Uranium refactor.

Reply:
My point is that over sharing data and overcoupling is a bad idea. Making it clear what is and isn't shared is desirable, even if you have more parameters. (See 'dependency injection')

Defining a clear application interface is a good also.